hip/solver/batch_bicgstab_kernels.hip.cpp

```diff
@@ -57,19 +74,189 @@ namespace hip {
 namespace batch_bicgstab {
 
 
+#include "common/cuda_hip/base/batch_multi_vector_kernels.hpp.inc"
+#include "common/cuda_hip/components/uninitialized_array.hpp.inc"
+#include "common/cuda_hip/matrix/batch_dense_kernels.hpp.inc"
+#include "common/cuda_hip/matrix/batch_ell_kernels.hpp.inc"
+#include "common/cuda_hip/solver/batch_bicgstab_kernels.hpp.inc"
+
+
+template <typename BatchMatrixType>
+int get_num_threads_per_block(std::shared_ptr<const DefaultExecutor> exec,
+                              const int num_rows)
+{
+    int nwarps = num_rows / 4;
+    if (nwarps < 2) {
+        nwarps = 2;
+    }
+    const int min_block_size = 2 * config::warp_size;
+    const int device_max_threads =
+        ((std::max(num_rows, min_block_size)) / config::warp_size) *
+        config::warp_size;
+    const int num_regs_used_per_thread = 64;
```

Comment:
why is it 64 here?

Reply:
That is the [value for maximum occupancy](https://gpuopen.com/learn/amd-lab-notes/amd-lab-notes-register-pressure-readme/). HIP does not have a function to query the number of registers for a given function like CUDA.

Reply:
Could you add a comment for that? it will be helpful for the others who wants to changes the parameters in the future.